src/decode/rr/draft_ietf_dnsop_svcb_https.rs

```diff
@@ -93,6 +44,67 @@ impl<'a, 'b: 'a> Decoder<'a, 'b> {
             https,
         })
     }
+
+    /// Decode a single service parameter
+    ///
+    /// Parameters:
+    /// - `service_parameter_key` - the IANA-controlled numeric identifier as defined in section
+    ///                             14.3 of the RFC
+    ///
+    /// Returns:
+    /// - `Ok(ServiceParameter)` - if there were no issues decoding the value
+    /// - `Err` - if there was any decoding error
+    fn rr_service_parameter(
+        &mut self,
+        service_parameter_key: u16,
+    ) -> DecodeResult<ServiceParameter> {
+        let service_parameter = match service_parameter_key {
+            0 => {
+                let mut key_ids = vec![];
+                while !self.is_finished()? {
+                    key_ids.push(self.u16()?);
+                }
+                ServiceParameter::MANDATORY { key_ids }
+            }
+            1 => {
+                let mut alpn_ids = vec![];
+                while !self.is_finished()? {
+                    alpn_ids.push(self.string()?);
+                }
+                ServiceParameter::ALPN { alpn_ids }
+            }
+            2 => ServiceParameter::NO_DEFAULT_ALPN,
+            3 => ServiceParameter::PORT { port: self.u16()? },
+            4 => {
+                let mut hints = vec![];
+                while !self.is_finished()? {
+                    hints.push(self.ipv4_addr()?);
+                }
+                ServiceParameter::IPV4_HINT { hints }
+            }
+            5 => {
+                // Note the RFC does not explicitly state that the length is two octets
+                // "In wire format, the value of the parameter is an ECHConfigList [ECH],
+                // including the redundant length prefix." - RFC Section 9
+                let _ = self.u16()?; // length
+                let config_list = self.vec()?;
```

Comment:
Could you add here a length check?
```rust
let lenght = self.u16()?;
let config_list = self.vec()?;
if lenght != config_list.len() {
    return Err(...);
}
```

Reply:
I implemented the length check. I added a new `DecodeError` variant called `ECHLengthMismatch`. I also removed a variant that I had mistakenly added before but forgot to remove.